### ISSUE TITLE
Constant cpu usage after quitting mixer 2.3.0+ on mac

### ISSUE THREAD

**Issue by serakeri** [pygame/pygame#3840]
**Environment:**
macOS M1, arm64 python 3.10.11
* **bug observed** on tests with pygame 2.3.0 and later
pygame 2.3.0 (sdl 2.24.2)
pygame 2.4.0 (sdl 2.26.4)

* **bug not observed** on versions of pygame before 2.3.0 (coreaudiod idle at/near 0% cpu usage)
pygame 2.1.2 (sdl 2.0.18)
pygame 2.1.3 (sdl 2.0.22)
pygame 2.2.0 (sdl 2.0.22)

Tested on an x86 mac too which did not use as much cpu (only about 3-5%) but it is likely the same bug. I was not able to reproduce the bug on windows since it is related to the mac process "coreaudiod" and windows continued to idle at 0% with nothing else running.

**Current behavior:**

Initializing then quitting the pygame.mixer uses a constant 10-15% cpu on the process **coreaudiod** and keeps the audio session open until the python script ends

**Expected behavior:**

Initializing then quitting the pygame.mixer uses negligible cpu on the process **coreaudiod** until the python script ends.

**Steps to reproduce:**

1. Open activity monitor and make sure the process "coreaudiod" is idle (0% cpu usage). Do not run other media apps so the cpu usage of "coreaudiod" can be isolated to pygame. If it's not at 0% cpu usage then quit other media apps until it is at 0%.
2. Run a python script that initializes pygame.mixer, quits it, then sleeps forever.
3. Open activity monitor and observe "coreaudiod" cpu usage at 10-15% until the script is closed. You can also see the audio session was not stopped until the script is closed manually with the steps in "Additional Info"

**Test code**

```
#!/usr/bin/env python3

import pygame
import time

pygame.mixer.init()
pygame.mixer.quit()

print("waiting...")

while True:
	time.sleep(10)
```
**Additional Info**

The audio session duration can be observed with the command
`log stream --process coreaudiod --info --debug `

There may be other relevant information in the coreaudiod log but I noticed the session duration differs on versions with and without the bug. On the versions with the bug the session length will be short as expected, e.g.

```
2023-05-06 17:22:02.571504-0400 0x30ea330  Default     0x0                  2368   0    coreaudiod: (libAudioStatistics.dylib) [com.apple.coreaudio:carc]      CAReportingClient.mm:537   Sending message { message="{
    "session_duration" = "0.08515393733978271";
}", reporters="(
    10170482557482
)" }
```

while versions that are bugged will have a session duration closer to how long the python script was left running even though the mixer was stopped immediately.

**Note: The coreaudiod session duration message above will not be displayed until the python script is closed**

**Comment by illume** [comment]
Hi. Thanks for the bug report. Especially for testing on different versions!

The next step is to bisect the commits, to try and find where the issue is.

I wonder if the issue is there with a [pygame compiled from source](https://www.pygame.org/wiki/MacCompile)?

```bash
brew install sdl2 sdl2_image sdl2_mixer sdl2_ttf pkg-config
python3 -m pip install git+https://github.com/pygame/pygame.git
```

Then to try different versions of mixer.c see the history for that file: https://github.com/pygame/pygame/commits/main/src_c/mixer.c

```bash
$ git log --oneline src_c/mixer.c
3b76aca7 Fix the memory leak a different way
da8c645b Fixed inconsistency in `tp_name`s, changed all to `pygame.module.Type`
435741eb Fix big endian bug in mixer get_init
40b8f9dd Fixed segault in subclass of Sound (#3492)
2d4bd89a Use METH_O where possible
98d06e7d Update SDL_mixer
d31eae2c Typos: get em out of here!
313be964 Simplify some boolean handling
6c82ef56 Merge branch 'pygame:main' into pygame-wasm-static
49607276 Raise only pygame.error when mixer loading fails
55c71c3d format
20d5689f oops in mixer init
bad4457f static: fixes more global names conflicts
6b1cb35d partial static build support for pygame (WIP)
10e89db7 Merge pull request #3080 from pmp-p/pygame-fix-fpc
fe9e421f Merge pull request #3067 from pygame/ankith26-type-stuct
35597a23 fix/norm function pointer calls for METH_NOARGS
8a1ce76e Fix more typestub warnings with dunders
981f4778 Use static dot declarations for type objects
28fce044 Update Event docs, use channel_init in pgChannel_New
4d73acc7 Remove joystick doc changes, reformat files
fa43cf81 Add init and new functions for channel, clock and event classes
e5f1f511 Merge branch 'main' into c-code-cleanups
29a87696 Merge pull request #2919 from pygame/ankith26-werror
c79a0df1 Move to preferred functions / macros
295338fc Fix most MSVC static analyzer warnings
7f836f12 Fix warnings, make warnings error on CI
f6d3d34f clean modinit code, fix bugs, use recommended API
025f40a1 lint: Running clang-format again on src_c
```
Maybe we go back to this previous mixer.c commit : 49607276 Raise only pygame.error when mixer loading fails

```bash
python3 -m pip install git+https://github.com/pygame/pygame.git@49607276
```


This will help us narrow down the issue. Is it a problem with a newer SDL version? A commit on mixer.c? Maybe a commit that doesn't touch mixer.c? It could also be a problem caused by a particular version of SDL2. Finally it might be a problem with how our binary wheel is compiled. With a few install/tests we should be able to figure out where the problem lies.

**Comment by serakeri** [comment]
Thank you for the helpful info! I built from source which revealed it's related to the newer versions of sdl and not the newer versions of pygame.

**versions built from source that displayed the bug**
pygame 2.5.0.dev1 (sdl 2.26.5)
pygame 2.2.0 (sdl 2.26.5)

**versions built from source that did not display the bug**
pygame 2.5.0.dev1 (sdl 2.0.22)
pygame 2.2.0 (sdl 2.0.22)

I tried for a bit but haven't found out how to build sdl2 from source in a way that also works with pygame so I can test where in SDL it broke. I think https://github.com/libsdl-org/SDL/commit/266ca2c933a308fe6ac21c29ce034637fb05e015 could be related but I have not been able to test or confirm it yet.

I will try again later to write a test case for SDL2 to see if I can reproduce it more simply that way unless there is an easy way to build sdl2 and pygame together from source.
